bot/exts/moderation/infraction/_utils.py

```diff
@@ -298,3 +300,26 @@ async def send_private_embed(user: MemberOrUser, embed: discord.Embed) -> bool:
             "The user either could not be retrieved or probably disabled their DMs."
         )
         return False
+
+
+class StaffBanConfirmationView(interactions.ViewWithUserAndRoleCheck):
+    """The confirmation view that is sent when a moderator attempts to ban a staff member."""
+
+    confirmed = False
```

Comment:
Shouldn't this be an instance attribute? Maybe this is fine if the instance is not re-used, but it's not clear this class's instances are not intended to be re-used nor would we have a good way to enforce that.

Reply:
I'm not sure if making it an instance attribute nets us anything as the confirmed flag wouldn't be reset to false either way? But sure, why not.

Reply:
I didn't correctly word what I was thinking. If it's a class attribute, then the state is persisted across all instances. This means `confirmed` will remain True for subsequent command invocations, right? 

You're correct that as an instance attribute, the instance cannot be re-used because it's never reset. But I think an instance attribute solves the above case.

Reply:
```python
ichard26@asus-ubuntu:~$ cat test.py
class A:
    flag = False

a = A()
a2 = A()
a.flag = True
print(a.flag, a2.flag)
ichard26@asus-ubuntu:~$ python test.py
True False
```
As far as I understand, unless the class attribute is directly modified via the class, state won't persist across all instances.

Reply:
Okay yeah, that makes sense now that I see it. Thanks for clearing that up - was not intuitive to me at a glance.